src/components/TimerPanel.js

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import classnames from 'classnames';
+
+const TimerPanel = ({ numberOfMinutes, numberOfSeconds, sessionNumber, markSessionAsComplete }) => {
+  // belows are local states(to this file), will only be used here, not global
+  const [minutes, setMinutes] = useState(numberOfMinutes);
+  const [seconds, setSeconds] = useState(numberOfSeconds);
+
+  // the 1st session never starts on its own
+  // if its is the end of all sessions, we go back to display the 1st session, without auto-starting
+  const checkIfLastSession = () => {
+    return sessionNumber > 0 ? true : false;
+  }
+  // per https://stackoverflow.com/questions/68945060/react-make-usestate-initial-value-conditional
+  // it is better to set a conditional initial state in a func, so it wont be evaluated on each component render
+  const [isActive, setIsActive] = useState(() => checkIfLastSession());
+
+  useEffect(() => {
+    let timeInterval;
+    if (isActive) {
+      timeInterval = setInterval(() => {
+        // if there are seconds left
+        if (seconds > 0) {
+          setSeconds(seconds - 1);
+        }
+        // if there are no seconds and mins left
+        if (seconds === 0) {
+          if (minutes === 0) {
+            setIsActive(false)
+            clearInterval(timeInterval)
+            // if seconds are at 0, but there are mins left
+          } else {
+            // at 0 seconds, minutes is -1 since it has started already
+            setMinutes(minutes - 1);
+            setSeconds(59);
+          }
+        }
+      }, 1000)
+    } else if (!isActive) {
+      clearInterval(timeInterval)
+    }
+    return () => clearInterval(timeInterval)
+  }, [isActive, seconds]);
+
+  useEffect(() => {
+    // session is complete, not paused
+    if (!isActive && minutes === 0 && seconds === 0) markSessionAsComplete();
```

Comment:
Don't use one liner like these. This is consider as bad practice as it makes it harder to read.
```suggestion
    if (!isActive && minutes === 0 && seconds === 0) {
      markSessionAsComplete();
    }
```
Also you need more than just `isActive` for this `useEffect`'s dependencies. You need to add `minutes`, `seconds`, and `markSessionAsComplete` as dependencies as well. Basically you need to add any function/variables that will be used in the effect.